### PR TITLE
Add support for isolate v2

### PR DESCRIPTION
**Pull request by minhnhatnoe**
On startup, check for the isolate daemon's status (or run the daemon as a goroutine if in docker container). Processes in the container's root cgroup will be moved to the subgroup /daemon by the custom isolate.

### CODE REVIEW

File: worker/sandbox/isolate/sandbox.go [hunk 1]
```diff
@@ -19,40 +20,73 @@ import (
...
 func init() {
+	if v, ok := os.LookupEnv("KJUDGE_ISOLATE_V1"); ok {
+		isolateCommandV1 = v
+	}
 	if v, ok := os.LookupEnv("KJUDGE_ISOLATE"); ok {
-		isolateCommand = v
+		isolateCommandV2 = v
+	}
+	if v, ok := os.LookupEnv("KJUDGE_ISOLATE_DAEMON"); ok {
+		isolateDaemonCommand = []string{v}
```
Comment (natsukagami) [inline worker/sandbox/isolate/sandbox.go:37]: This looks suspiciously wrong :thinking: basically this means we should not be able to pass anything with arguments.

I think the better way to do this is to assume `systemd` and only allow customizing the service name.

File: worker/sandbox/sandbox.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ import (
 // and "raw" (NOT RECOMMENDED, RUN AT YOUR OWN RISK).
 // Which sandbox is used can be set at runtime with a command-line switch.
 type Runner interface {
+	Start()
```
Comment (natsukagami) [inline worker/sandbox/sandbox.go:19]: No `Stop`? :thinking:

File: worker/sandbox/isolate/sandbox.go (hunk 1 above)
Comment (minhnhatnoe) [reply worker/sandbox/isolate/sandbox.go:37]: In Docker containers, systemd won't be installed, so KJUDGE_ISOLATE_DAEMON should be able to receive any command. I'm going to make it execute any passed command with `/bin/sh -c`

File: worker/sandbox/sandbox.go (hunk 2 above)
Comment (minhnhatnoe) [reply worker/sandbox/sandbox.go:19]: Added

**Comment by minhnhatnoe** [comment]
You might also want to merge changes from `master` branch at `https://github.com/minhnhatnoe/kjudge.wiki.git`. I have added notes about using `--cap-add` instead of `--privileged`.

**Comment by natsukagami** [comment]
Think we can YOLO this one for now ^^

File: docker/Dockerfile [hunk 3]
```diff
@@ -51,7 +38,7 @@ RUN scripts/install_tools.sh
 RUN go generate && go build -tags production -o kjudge cmd/kjudge/main.go
 
 # Stage 3: Create awesome output image
-FROM base-ubuntu
+FROM ghcr.io/minhnhatnoe/isolate:v2.1.5
```
Comment (natsukagami) [inline docker/Dockerfile:41]: TODO put this somewhere centralized alongside other versioned stuff

**Review by natsukagami — APPROVED** [review]
:yolo:

**Pull request merged**